front/components/data_source/DocumentOrTableUploadOrEditModal.tsx

```diff
@@ -0,0 +1,532 @@
+            {isTable ? (
+              <div>
+                <Page.SectionHeader
+                  title="Description"
+                  description="Describe the content of your CSV file. It will be used by the LLM model to generate relevant queries."
+                />
+                <textarea
+                  name="table-description"
+                  placeholder="This table contains..."
+                  rows={10}
+                  value={tableOrDoc.description}
+                  onChange={(e) =>
+                    setTableOrDoc((prev) => ({
+                      ...prev,
+                      description: e.target.value,
+                    }))
+                  }
+                  className={classNames(
+                    "font-mono text-normal block w-full min-w-0 flex-1 rounded-md",
+                    "border-structure-200 bg-structure-50",
+                    "focus:border-action-300 focus:ring-action-300"
+                  )}
+                />
+              </div>
+            ) : (
+              <div>
+                <Page.SectionHeader
+                  title="Associated URL"
+                  description="The URL of the associated document (if any). Will be used to link users to the original document in assistants citations."
+                />
+                <Input
+                  placeholder="https://..."
+                  name="sourceUrl"
+                  value={tableOrDoc.sourceUrl}
+                  onChange={(value) =>
+                    setTableOrDoc((prev) => ({ ...prev, sourceUrl: value }))
+                  }
+                />
+              </div>
+            )}
+
+            <div>
+              <Page.SectionHeader
+                title={isTable ? "CSV File" : "Text content"}
+                description={
+                  isTable
+                    ? "Select the CSV file for data extraction. The maximum file size allowed is 50MB."
+                    : `Copy paste content or upload a file (text or PDF). Up to ${
+                        plan.limits.dataSources.documents.sizeMb === -1
+                          ? "2"
+                          : plan.limits.dataSources.documents.sizeMb
+                      } MB of raw text.`
+                }
+                action={{
+                  label: (() => {
+                    if (uploading) {
+                      return "Uploading...";
+                    } else if (tableOrDoc.file) {
+                      return tableOrDoc.file.name;
+                    } else if (initialId) {
+                      return "Replace file";
+                    } else {
+                      return "Upload file";
+                    }
+                  })(),
+                  variant: "primary",
+                  icon: DocumentPlusIcon,
+                  onClick: () => fileInputRef.current?.click(),
+                }}
+              />
+              <input
+                type="file"
+                ref={fileInputRef}
```

Comment:
Would it be possible to reset the input when closing the modal ? It was not the case before but it's quite cumbersome to still have the previous value when reopening the modal (after a successful upload or a cancel )